services/jenkins/jenkins-plugin-installs.service.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+const Joi = require('joi')
+
+const BaseJsonService = require('../base-json')
+const { NotFound, InvalidParameter } = require('../errors')
+const {
+  downloadCount: downloadCountColor,
+} = require('../../lib/color-formatters')
+const { metric } = require('../../lib/text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schemaInstallations = Joi.object()
+  .keys({
+    installations: Joi.object()
+      .required()
+      .pattern(nonNegativeInteger, nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+const schemaInstallationsPerVersion = Joi.object()
+  .keys({
+    installationsPerVersion: Joi.object()
+      .required()
+      .pattern(Joi.string(), nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+class JenkinsPluginInstalls extends BaseJsonService {
+  async fetch({ plugin, version }) {
+    const url = `https://stats.jenkins.io/plugin-installation-trend/${plugin}.stats.json`
+    const schema = this.constructor._getSchema(version)
+    return this._requestJson({ url, schema })
+  }
+
+  static render({ label, installs }) {
+    return {
+      label: label,
+      message: metric(installs),
+      color: downloadCountColor(installs),
+    }
+  }
+
+  static _getSchema(version) {
+    if (version) {
+      return schemaInstallationsPerVersion
+    } else {
+      return schemaInstallations
+    }
+  }
+
+  static _getLabel(version) {
+    if (version) {
+      return 'installs@' + version
+    } else {
+      return 'installs'
+    }
+  }
+
+  async handle({ info, plugin, version }) {
+    const label = this.constructor._getLabel(version)
+    const json = await this.fetch({ plugin, version })
+
+    let installs
+    if (info === 'iv') {
+      installs = json.installationsPerVersion[version]
+      if (!installs) {
+        throw new NotFound({
+          underlyingError: new Error('non-existent version'),
```

Comment:
By specifying the `prettyMessage` property, I believe we could be a bit more specific in the badge error message and display "version not found" for instance.

For the total installs, we could specify the 404 response code message by adding a parameter similar to the following to requestJSON:
`````
errorMessages: {
    404: 'plugin not found',
},

Reply:
I was wondering why `non-existent version` from the `underlyingError` didn't show up, but didn't bother to look into it since that's how the `GemDownloads` service [is doing it as well](https://github.com/badges/shields/blob/454575c4cbc8a736a597f878b9afbb2fd8122a75/services/gem/gem-downloads.service.js#L91).